### ISSUE TITLE
[BUG] Iterating over result of BlobServiceClient.listBlobContainers results in infinite loop

### ISSUE THREAD

**Issue by willemv** [Azure/azure-sdk-for-java#18881]
**Describe the bug**
Calling `BlobServiceClient.listBlobContainers` and iterating over the results endlessly loops over all containers

**To Reproduce**
* Set up an Azure Storage account
* Add a limited amount of blob containers (e.g. two)
* create a `BlobServiceClient`
* call `listBlobContainers` on the client, and iterate over the result

***Code Snippet***
Using `stream()`
``` 
BlobServiceClient azureClient = new BlobServiceClientBuilder()
        .connectionString(connectionString)
        .buildClient();
azureClient.listBlobContainers()
           .stream()
           .forEach(blobContainerItem -> System.out.println("Container: " + blobContainerItem.getName()));
```

Using the for-each construct:
```
for (BlobContainerItem blobContainerItem : azureClient.listBlobContainers()) {
    System.out.println("Container: " + blobContainerItem.getName());
}
```

Using the paged functionality:
```
for (PagedResponse<BlobContainerItem> page : azureClient.listBlobContainers().iterableByPage(100)) {
    for (BlobContainerItem blobContainerItem : page.getValue()) {
        System.out.println("Container: " + blobContainerItem.getName());
    }
}
```

**Expected behavior**
The name of each available share is printed once

Actual result: this keeps printing the same share names over and over, without ever stopping.

**Setup (please complete the following information):**
 - OS: Linux, macOS, Windows
 - IDE : IntelliJ, Eclipse, gradle, maven, ...
 - Version of the Library used: 
    - `azure-storage-blob` : `12.10.0`
    - `azure-storage-common` : `12.10.0`
    - `azure-core`: `1.12.0`

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by rickle-msft** [comment]
Hi, @willemv. Thanks for reporting this issue. Can you please take a look at #9465 and see if any of the solutions there might apply to your scenario? Specifically at the end of the thread we concluded in that case it was either a classloader or dependency issue depending on the scenario.

**Comment by willemv** [comment]
Hi @rickle-msft , #9465 looks like our bug indeed. I know we have jackson in our classpath. I will investigate and see if the mentioned workarounds work for us.

**Comment by willemv** [comment]
@rickle-msft #9465 looks like our bug, but the workarounds don't work for us.

The problem starts when you have a more recent version of jackson in the classpath than the one that the azure java sdk uses by default.

I.e., by default the azure sdk uses version '2.11.3'. In our project we need a more recent version. At the time of writing, the most recent version of jackson is "2.12.1". If we use that version, you get the infinite loop.

I've attached a sample gradle project with unit tests that you can open in typical IDE's, so you can see what I did to reproduce this : [azure-bug-reproduction.zip](https://github.com/Azure/azure-sdk-for-java/files/5973768/azure-bug-reproduction.zip)

This project allows you to switch between jackson version 2.11.3 and 2.12.1 by adjusting the `build.gradle` file. It has a series of unit tests that show the problem with jackson version 2.12.1. You can specify the ConnectionString to use when running this test by setting the necessary environment variables (or simply editing the source files of course).

This also contains unit tests for another problem I encountered: `ShareServiceClient#listShares()` and `BlobServiceClient#listBlobContainers()` both throw an exception when no shares or blob containers are present in the account. I'll log a separate issue for that, but I didn't feel like creating a separate gradle project for that.

**Comment by rickle-msft** [comment]
@willemv I recently found out that there's actually work underway to upgrade our jackson dependency, so I think this should be resolved for you soon: https://github.com/Azure/azure-sdk-for-java/pull/18884

**Comment by gapra-msft** [comment]
This issue will be resolved once the 12.11.0-beta.2 azure-storage-blob library is released

**Comment by gapra-msft** [comment]
Hi @willemv, we recently released azure-storage-blob 12.11.0-beta.2 that should fix this issue. Please try it out and let us know if you are still hitting the issue.

Feel free to close this once you've confirmed the latest release fixes your issue.

**Comment by willemv** [comment]
The issue is fixed in 12.11.10-beta.2 and later, thanks
